.circleci/config.yml

```diff
@@ -316,3 +357,33 @@ workflows:
     - ipfs-webui:
         requires:
           - build
+    - docker-build
+    - docker-push:
+        requires:
+        - docker-build
+        - golint
+        - gotest
+        - sharness
+        - interop
+        - go-ipfs-api
+        - go-ipfs-http-client
+        - ipfs-webui
+
+  # NOTE: You have to opt in to get circleci to trigger builds for a tat. As it is required that you set the tag filter to all dependent jobs, this is pulled out into a seperate workflow.
+  # see: https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag
+  docker-on-tag: 
+    jobs:
+      - docker-build:
+          filters:
+            tags:
+              only: /^v[0-9].*|^cluster.+/
+            branches:
+              ignore: /.*/
+      - docker-push:
+          requires:
+            - docker-build
+          filters:
+            tags:
+              only: /^v[0-9].*|^cluster.+/
+            branches:
+              ignore: /.*/
```

Comment:
YAML anchors might be helpful here:

```yaml
buildable-tag-filter: &buildable-tag-filter
  tags:
    only: /^v[0-9].*|^cluster.+/
  branches:
    ignore: /.*/

workflows:
  docker-on-tag:
  jobs:
    - docker-build:
        filters: *buildable-tag-filter
    - docker-push:
        requires:
          - docker-build
        filters: *buildable-tag-filter
```

Reply:
Actually, I think this would have the same result is slightly more concise:

```yaml
only-buildable-tags: &only-buildable-tags
  filters:
    tags:
      only: /^v[0-9].*|^cluster.+/
    branches:
      ignore: /.*/

workflows:
  docker-on-tag:
    jobs:
      - docker-build: *only-buildable-tags
      - docker-push:
          <<: *only-buildable-tags
          requires:
            - docker-build
```